akinet/net_traffic.go

```diff
@@ -44,6 +45,61 @@ func (rb RawBytes) String() string {
 	return memview.MemView(rb).String()
 }
 
+// Represents metadata from an observed TCP packet.
+type TCPPacketMetadata struct {
+	// Uniquely identifies a TCP connection.
+	ConnectionID akid.ConnectionID
+
+	// Whether the SYN flag was set in the observed packet.
+	SYN bool
+
+	// Whether the ACK flag was set in the observed packet.
+	ACK bool
+
+	// Whether the FIN flag was set in the observed packet.
+	FIN bool
+
+	// Whether the RST flag was set in the observed packet.
+	RST bool
```

Comment:
I wonder whether we are going to be happier in the long run if we just mirror all the TCP flag bits as a single field.

🤷  This seems fine, we can change it later.

Reply:
Yeah, not sure. They're already broken out into separate fields by the time they reach us.